SpatialGDK/Source/SpatialGDKTests/SpatialGDK/LoadBalancing/LayeredLBStrategy/LayeredLBStrategyTest.cpp

```diff
@@ -0,0 +1,430 @@
+		FString::Printf(TEXT("Who Should Have Authority. Actual: %d, Expected: %d"), Actual, Expected),
+		Actual, Expected);
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_THREE_PARAMETER(FCheckMinimumWorkers, TSharedPtr<TestData>, TestData, FAutomationTestBase*, Test, uint32, Expected);
+bool FCheckMinimumWorkers::Update()
+{
+	const uint32 Actual = TestData->Strat->GetMinimumRequiredWorkers();
+	Test->TestEqual(
+		FString::Printf(TEXT("Strategy for minimum required workers. Actual: %d, Expected: %d"), Actual, Expected),
+		Actual, Expected);
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_THREE_PARAMETER(FCheckStratIsReady, TSharedPtr<TestData>, TestData, FAutomationTestBase*, Test, bool, Expected);
+bool FCheckStratIsReady::Update()
+{
+	const auto& Strat = TestData->Strat;
+	Test->TestEqual(
+		FString::Printf(TEXT("Strategy for minimum required workers. Actual: %d, Expected: %d"), Strat->IsReady(), Expected),
+		Strat->IsReady(), Expected);
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_THREE_PARAMETER(FSpawnLayer1PawnAtLocation, TSharedPtr<TestData>, TestData, FName, Handle,
+	FVector, Location);
+bool FSpawnLayer1PawnAtLocation::Update()
+{
+	FActorSpawnParameters SpawnParams;
+	SpawnParams.bNoFail = true;
+	SpawnParams.SpawnCollisionHandlingOverride = ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
+
+	AActor* NewActor = TestData->TestWorld->SpawnActor<ALayer1Pawn>(Location, FRotator::ZeroRotator, SpawnParams);
+	TestData->TestActors.Add(Handle, NewActor);
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_THREE_PARAMETER(FSpawnLayer2PawnAtLocation, TSharedPtr<TestData>, TestData,
+	FName, Handle, FVector, Location);
+bool FSpawnLayer2PawnAtLocation::Update()
+{
+	FActorSpawnParameters SpawnParams;
+	SpawnParams.bNoFail = true;
+	SpawnParams.SpawnCollisionHandlingOverride = ESpawnActorCollisionHandlingMethod::AlwaysSpawn;
+
+	AActor* NewActor = TestData->TestWorld->SpawnActor<ALayer2Pawn>(Location, FRotator::ZeroRotator, SpawnParams);
+	TestData->TestActors.Add(Handle, NewActor);
+
+	return true;
+}
+
+DEFINE_LATENT_AUTOMATION_COMMAND_FIVE_PARAMETER(FCheckActorsAuth, TSharedPtr<TestData>, TestData, FAutomationTestBase*, Test, FName, FirstActorName, FName, SecondActorName, bool, ExpectEqual);
+bool FCheckActorsAuth::Update()
+{
+	const auto& Strat = TestData->Strat;
+	const auto& TestActors = TestData->TestActors;
+
+	const VirtualWorkerId FirstActorAuth = Strat->WhoShouldHaveAuthority(*TestActors[FirstActorName]);
+	const VirtualWorkerId SecondActorAuth = Strat->WhoShouldHaveAuthority(*TestActors[SecondActorName]);
+
+	if (ExpectEqual)
+	{
+		Test->TestEqual(
+			FString::Printf(TEXT("Actors should have the same auth. Actor1: %d, Actor2: %d"), FirstActorAuth, SecondActorAuth),
+			FirstActorAuth, SecondActorAuth);
+	}
+	else {
+		Test->TestNotEqual(
+			FString::Printf(TEXT("Actors should have different auth. Actor1: %d, Actor2: %d"), FirstActorAuth, SecondActorAuth),
+			FirstActorAuth, SecondActorAuth);
+	}
```

Comment:
You could simplify into one ExpectEqual between `(FirstActorAuth==SscondActorAuth) == ExpectEqual`?

Reply:
ye i get that, but then i would have to do conditional text formatting on the message